### PR TITLE
make ringtone and uplink code 6 notes long

### PR DESCRIPTION
**Pull request by BasedUser**
## About the PR
literally just makes uplinks 2 notes longer
oh and also uplinks can include sharps now

this PR resolves the issue of skids being able to bruteforce through 625 (5\*\*4) combinations in 5 minutes, now they'll have to bruteforce through 3M (12\*\*6) in 16 days (unless they bot the attack in which case we wait for packet flood bans) or literally ask the traitor involved for their code

originally coded for a downstream, bumped because this issue is more real than surgery

**Media**
image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: router
- tweak: Ringtones now have 2 more notes you can play around with.
- fix: The Syndicate has taken advantage of this hardware expansion, making brute-force attacks on their uplinks unfeasible.

### CODE REVIEW

**Comment by Myakot** [comment]
I honestly can't believe somebody is that dumb that they decided to use bots for brute-forcing a pda in RP setting.

**Comment by asperger-sind** [comment]
> I honestly can't believe somebody is that dumb that they decided to use bots for brute-forcing a pda in RP setting.

A lot of the playerbase views the game in a more gameplay setting rather than an RP setting, so it's only understandable.

**Comment by BasedUser** [comment]
> I honestly can't believe somebody is that dumb that they decided to use bots for brute-forcing a pda in RP setting.

625 isn't really a large number either, you can go through it by hand as well if you're willing to spend 30 minutes of downtime, being generous

File: Content.Server/PDA/Ringer/RingerSystem.cs [hunk 1]
```diff
@@ -108,7 +108,30 @@ public void RandomizeRingtone(EntityUid uid, RingerComponent ringer, MapInitEven
...
+            };
+
+            var ringtone = new Note[RingtoneLength];
+
+            for (var i = 0; i < RingtoneLength; i++)
+            {
+                ringtone[i] = _random.Pick(notes);
+            }
+
+            uplink.Code = ringtone;
```
Comment (AJCM-git) [inline Content.Server/PDA/Ringer/RingerSystem.cs:134]: this is just copypaste from GenerateRingtone, why not simply modify it?

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Server/PDA/Ringer/RingerSystem.cs (hunk 1 above)
Comment (deltanedas) [reply Content.Server/PDA/Ringer/RingerSystem.cs:134]: modification being either bool sharp or Note[] notes

Comment (BasedUser) [reply Content.Server/PDA/Ringer/RingerSystem.cs:134]: ...as above

there's currently only 2 usecases, and that's standard ringtone generation and uplink code generation, which means you have a bool of sus and not sus already

could probably do that instead but need to include both of the lists anyway and IMO is probably better kept explicit (being *right next to each other*)

File: Content.Client/PDA/Ringer/RingerBoundUserInterface.cs [hunk 2]
```diff
@@ -41,7 +41,7 @@ private bool TryGetRingtone(out Note[] ringtone)
                 return false;
             }
 
-            ringtone = new Note[4];
+            ringtone = new Note[6];
```
Comment (metalgearsloth) [inline Content.Client/PDA/Ringer/RingerBoundUserInterface.cs:44]: Use the constant for it.

File: Content.Server/PDA/Ringer/RingerSystem.cs [hunk 3]
```diff
@@ -182,7 +205,7 @@ public bool ToggleRingerUI(EntityUid uid, IPlayerSession session)
 
                 ringer.NoteCount++;
 
-                if (ringer.NoteCount > 3)
+                if (ringer.NoteCount > RingtoneLength-1)
```
Comment (metalgearsloth) [inline Content.Server/PDA/Ringer/RingerSystem.cs:208]:
```suggestion
                if (ringer.NoteCount > RingtoneLength - 1)
```

File: Content.Server/PDA/Ringer/RingerSystem.cs (hunk 1 above)
Comment (metalgearsloth) [reply Content.Server/PDA/Ringer/RingerSystem.cs:134]: Why isn't this just using a pentatonic anyway

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**